File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        int num = 0;
        ListNode* temp = head;
        while(temp != NULL){
            temp = temp->next;
            num++;
        }
        temp = head;
        int i = 0;
    
        while(i < num - n - 1){
            temp = temp->next;
            i++;
        }
        if(num - n == 0){
            return head->next;
        }
        if(temp->next == NULL){
            temp->next = NULL;
        }
        temp->next = temp->next->next;
        return head;
    }
};

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        count = 0
        curr = head
        while curr:
            count += 1
            curr = curr.next
        curr = head
        i = 1
        if (count - n) == 0:
            head = head.next
            return head
        while i < (count - n):
            curr = curr.next
            i = i + 1
        temp = curr.next
        curr.next = temp.next
        return head


